### PR TITLE
release/1.7.0 - site config and doc updates

### LINKED ISSUE #1020: [INSTALL]: ip@5
### Package name

ip

### Package version/tag

5.0.0

### Build options

(existing) +deprecated

### Installation timeframe

1.7.0

### Other information

sp no longer needed; dependent recipes will need updating

### LINKED ISSUE #1042: spack-stack 1.7.0 testing
This is a tracking issue for testing release/1.7.0.

For now, to test out the shortened naming schemes, let's use, e.g., ue-intel, ue-gcc, etc. (and remember to install a separate stack for each compiler).

Systems:
- [x] Hercules (@RatkoVasic-NOAA)
/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.7.0/envs/ (gsi-addon-gcc  gsi-addon-intel  ue-gcc  ue-intel)
- [x] Orion (@RatkoVasic-NOAA) 
- [x] Discover (@climbfuji and @srherbener)
- [x] Casper (@climbfuji)
- [x] Derecho (@RatkoVasic-NOAA) 
/glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.7.0/envs (ue-gcc  ue-intel)
- [x] Acorn (@AlexanderRichert-NOAA; /lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.7.0/envs/ue-intel{19,2022})
- [x] Gaea C5 (@RatkoVasic-NOAA)
/ncrc/proj/epic/spack-stack/spack-stack-1.7.0/envs (gsi-addon-intel  ue-intel)
- [x] Hera (@AlexanderRichert-NOAA; /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.7.0/envs/{ue,gsi-addon}-{intel,gcc}; ran a couple of UFS test builds)
- [x] Jet (@AlexanderRichert-NOAA;  /mnt/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.7.0/envs/{ue,gsi-addon}-{intel,gcc})
- [x] Narwhal (@climbfuji)
- [x] Nautilus (@climbfuji)
- [x] S4 (@climbfuji and @ashley314)
- [x] AWS AMI RH 8 (@climbfuji)
- [x] AWS CI cluster (existing, c6i; @climbfuji)
- [ ] ~~AWS ParallelCluster~~ (decommissioned - if resurrected, then @climbfuji)
- [x] NOAA ParallelWorks EPIC (@natalie-perlin) 
AWS, GCP, Azure: /contrib/spack-stack/spack-stack/1.7.0/envs/ue-intel
NB: Azure built a spack-stack directly in common space, launching "spack install" twice.
- [x] NOAA ParallelWorks JCSDA (@climbfuji)

Issues encountered:
 - RESOLVED: Hera needs zlib-ng to be static only. I have added the necessary config changes (incl. update to compiler OS and external OpenMPI module/version).
 - Note on UFS Weather Model: If testing against UFS Weather Model, note that a handful of changes are needed on account of ip@5. Message me or run a `git diff` in /scratch1/NCEPDEV/nems/Alexander.Richert/ufs-weather-model to see the changes needed (basically, `sp` becomes `ip` in all the CMakeLists.txt's, plus a no-openmp flag needs to be explicitly set in FV3/ccpp/physics/CMakeLists.txt.
 - RESOLVED: On Jet, py-nodeenv and awscli-v2 were failing to build because they can't load `ssl_` from urllib3. Resolved by using gcc/9.2.0b. 
 - Note: grib-util 1.4.0 won't build with older Intel compilers that lack F2018 standard support. If this is a problem anywhere other than Acorn, then let's implement a patch. For now this is fixed manually on Acorn, and will be fixed in the next version.
 - Note: To get odc to compile on Acorn, I had to patch the root CMakeLists.txt to use c++17 instead of c++11. Not sure why this isn't an issue on other systems.
 - RESOLVED: Hera: Running out of space for gsi-addon builds. EPIC has cleared out some space (thank you!).
 - RESOLVED: Jet: HDF5 failing to build with gcc due to using the centos7 gcc. Jet site config has been updated locally, permanent updates will be pushed shortly.

### LINKED ISSUE #1055: Problems of zlib-ng with intel@2021.5.0 on some - not all - systems
**Describe the bug**

We have noted problems on some - not all - platforms with ``intel@2021.5.0`` when we switched from ``zlib`` to ``zlib-ng`` in spack-stack-1.7.0. These issues went away when using a different version of the compiler (anything between 2021.3.0 and 2021.11.0).

One problem users have reported is errors when using `git clone`:
```
fatal: The remote end hung up unexpectedly
```
Another problem seems to be running `pip install`.

It is therefore recommended to avoid using ``intel@2021.5.0`` unless it is the only option - add this to our compiler compatibility table.

**To Reproduce**
On Discover SCU16 or S4, build unified environment for spack-stack-1.7.0 with `intel@2021.5.0` and try to run `git clone`

**Expected behavior**
None of the errors described above

**System:**
Discover SCU16 and S4 (these are the one we know are affected), potentially others

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

Update site configs and documentation for all platforms supported in spack-stack-1.7.0

### Testing

Tested all applications selectively/manually on all systems

### Applications affected

All

### Systems affected

All

### Dependencies

n/a

### Issue(s) addressed

Resolves https://github.com/JCSDA/spack-stack/issues/1055 by adding a note to avoid using `intel@2021.5.0`.

Resolves https://github.com/JCSDA/spack-stack/issues/1042

Resolves https://github.com/JCSDA/spack-stack/issues/1020

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [ ] ~~All dependency PRs/issues have been resolved and this PR can be merged.~~

### CODE REVIEW

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
Waiting for Orion for the last updates, then we can merge. @RatkoVasic-NOAA any updates? Thanks for doing those installs!

**Comment by RatkoVasic-NOAA** [comment]
@climbfuji , yes, there are good and bad news. Good news, GNU is installed, both unified environment and gsi addon:
```
/work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.7.0/envs/gsi-addon-gcc-centos/install/modulefiles/Core
/work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.7.0/envs/ue-gcc-centos/install/modulefiles/Core
```
Bad news, intel failed with "Internal compiler error", compiler intel/2018.4
/work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.7.0/envs/ue-intel-centos

**Comment by climbfuji** [comment]
I don't think we should be building with intel@2018? Can you remove that rogue compiler entry and just build with the correct one? intel@2022.0.2 - please?

**Comment by RatkoVasic-NOAA** [comment]
> I don't think we should be building with intel@2018? Can you remove that rogue compiler entry and just build with the correct one? intel@2022.0.2 - please?

@climbfuji, it worked for unified environment! GSI addon intel is underway. No problems are expected right now, so when finished, PR will follow.

File: doc/source/PreConfiguredSites.rst [hunk 1]
```diff
@@ -88,31 +98,29 @@ The following is required for building new spack environments and for using spac
...
    module load stack-python/3.10.13
-   module available
 
-For ``spack-stack-1.6.0`` with GNU, load the following modules after loading miniconda and ecflow:
+For ``spack-stack-1.7.0`` with GNU, load the following modules after loading miniconda and ecflow:
 
 .. code-block:: console
 
-   module use /work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.6.0/envs/unified-env/install/modulefiles/Core
+   module use /work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.6.0/envs/ue-gcc-centos/install/modulefiles/Core
```
Comment (fcvdb) [inline doc/source/PreConfiguredSites.rst]: stack-1.7.0?

File: doc/source/PreConfiguredSites.rst [hunk 2]
```diff
@@ -88,31 +98,29 @@ The following is required for building new spack environments and for using spac
...
    module load stack-gcc/10.2.0
    module load stack-openmpi/4.0.4
    module load stack-python/3.10.13
```
Comment (fcvdb) [inline doc/source/PreConfiguredSites.rst:115]: module load sp ?

File: doc/source/PreConfiguredSites.rst [hunk 3]
```diff
@@ -88,31 +96,29 @@ The following is required for building new spack environments and for using spac
...
-For ``spack-stack-1.6.0`` with GNU, load the following modules after loading miniconda and ecflow:
+For ``spack-stack-1.7.0`` with GNU, load the following modules after loading miniconda and ecflow:
 
 .. code-block:: console
 
-   module use /work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.6.0/envs/unified-env/install/modulefiles/Core
+   module use /work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.7.0/envs/ue-gcc-centos/install/modulefiles/Core
    module load stack-gcc/10.2.0
    module load stack-openmpi/4.0.4
    module load stack-python/3.10.13
```
Comment (fcvdb) [inline doc/source/PreConfiguredSites.rst:115]: module load  sp ?

Comment (climbfuji) [reply doc/source/PreConfiguredSites.rst:115]: NOT in the spack-stack documentation. This goes AFTER module load jedi-fv3-env ... in the jedi-docs

**Review by fcvdb — APPROVED** [review]
Thanks Dom!

**Comment by climbfuji** [comment]
I am going to merge this, understanding that there may be a follow-up PR with site config updates for systems that have mapl built without variants pflogger and extdata2g for Intel compiler versions 2021.7.0 and later.

**Pull request merged**